File: src/pt/joaomneto/titancompanion/adventure/impl/FFAdventure.java
package pt.joaomneto.titancompanion.adventure.impl;

import java.io.BufferedWriter;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import pt.joaomneto.titancompanion.R;
import pt.joaomneto.titancompanion.adventure.Adventure;
import android.os.Bundle;
import android.view.Menu;

public class FFAdventure extends Adventure {

	private int currentFirepower = -1;
	private int currentArmour = -1;
	private int initialFirepower = -1;
	private int initialArmour = -1;
	private int rockets = -1;
	private int ironSpikes = -1;
	private int oilCannisters = -1;
	private int spareWheels = -1;

	List<String> carEnhancements = new ArrayList<String>();

	protected static final int FRAGMENT_VEHICLE_COMBAT = 2;
	protected static final int FRAGMENT_VEHICLE_EQUIPMENT = 3;
	protected static final int FRAGMENT_EQUIPMENT = 4;
	protected static final int FRAGMENT_NOTES = 5;

	public FFAdventure() {
		super();
		fragmentConfiguration.clear();
		fragmentConfiguration.put(FRAGMENT_VITAL_STATS, new AdventureFragmentRunner(R.string.vitalStats, "pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureVitalStatsFragment"));
		fragmentConfiguration.put(FRAGMENT_COMBAT, new AdventureFragmentRunner(R.string.fights, "pt.joaomneto.titancompanion.adventure.impl.fragments.ff.FFAdventureCombatFragment"));
		fragmentConfiguration.put(FRAGMENT_VEHICLE_COMBAT, new AdventureFragmentRunner(R.string.vehicleCombat, "pt.joaomneto.titancompanion.adventure.impl.fragments.ff.FFVehicleCombatFragment"));
		fragmentConfiguration.put(FRAGMENT_VEHICLE_EQUIPMENT, new AdventureFragmentRunner(R.string.vehicleCombat, "pt.joaomneto.titancompanion.adventure.impl.fragments.ff.FFVehicleStatsFragment"));
		fragmentConfiguration.put(FRAGMENT_EQUIPMENT, new AdventureFragmentRunner(R.string.goldEquipment, "pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureEquipmentFragment"));
		fragmentConfiguration.put(FRAGMENT_NOTES, new AdventureFragmentRunner(R.string.notes, "pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureNotesFragment"));
	}

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		try {
			super.onCreate(savedInstanceState);

		} catch (Exception e) {
			e.printStackTrace();
		}

	}

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		getMenuInflater().inflate(R.menu.adventure, menu);
		return true;
	}

	@Override
	public void storeAdventureSpecificValuesInFile(BufferedWriter bw) throws IOException {

		bw.write("currentFirepower=" + currentFirepower + "\n");
		bw.write("currentArmour=" + currentArmour + "\n");
		bw.write("initialFirepower=" + initialFirepower + "\n");
		bw.write("initialArmour=" + initialArmour + "\n");
		bw.write("rockets=" + rockets + "\n");
		bw.write("ironSpikes=" + ironSpikes + "\n");
		bw.write("oilCannisters=" + oilCannisters + "\n");
		bw.write("spareWheels=" + spareWheels + "\n");

		bw.write("gold=" + getGold() + "\n");
		bw.write("provisions=" + getProvisions() + "\n");
		bw.write("provisionsValue=" + getProvisionsValue() + "\n");
		bw.write("carEnhancements=" + arrayToString(carEnhancements));
	}

	@Override
	protected void loadAdventureSpecificValuesFromFile() {

		setCurrentFirepower(Integer.valueOf(getSavedGame().getProperty("currentFirepower")));
		setCurrentArmour(Integer.valueOf(getSavedGame().getProperty("currentArmour")));
		setInitialFirepower(Integer.valueOf(getSavedGame().getProperty("initialFirepower")));
		setInitialArmour(Integer.valueOf(getSavedGame().getProperty("initialArmour")));
		setRockets(Integer.valueOf(getSavedGame().getProperty("rockets")));
		setIronSpikes(Integer.valueOf(getSavedGame().getProperty("ironSpikes")));
		setOilCannisters(Integer.valueOf(getSavedGame().getProperty("oilCannisters")));
		setGold(Integer.valueOf(getSavedGame().getProperty("gold")));
		setProvisions(Integer.valueOf(getSavedGame().getProperty("provisions")));
		setProvisionsValue(Integer.valueOf(getSavedGame().getProperty("provisionsValue")));
		setSpareWheels(Integer.valueOf(getSavedGame().getProperty("spareWheels")));
		setCarEnhancements(stringToArray(getSavedGame().getProperty("provisionsValue")));
	}

	public int getCurrentFirepower() {
		return currentFirepower;
	}

	public void setCurrentFirepower(int currentFirepower) {
		this.currentFirepower = currentFirepower;
	}

	public int getCurrentArmour() {
		return currentArmour;
	}

	public void setCurrentArmour(int currentArmour) {
		this.currentArmour = currentArmour;
	}

	public int getInitialFirepower() {
		return initialFirepower;
	}

	public void setInitialFirepower(int initialFirepower) {
		this.initialFirepower = initialFirepower;
	}

	public int getInitialArmour() {
		return initialArmour;
	}

	public void setInitialArmour(int initialArmour) {
		this.initialArmour = initialArmour;
	}

	public int getRockets() {
		return rockets;
	}

	public void setRockets(int rockets) {
		this.rockets = rockets;
	}

	public int getIronSpikes() {
		return ironSpikes;
	}

	public void setIronSpikes(int ironSpikes) {
		this.ironSpikes = ironSpikes;
	}

	public int getOilCannisters() {
		return oilCannisters;
	}

	public void setOilCannisters(int oilCannisters) {
		this.oilCannisters = oilCannisters;
	}

	public int getSpareWheels() {
		return spareWheels;
	}

	public void setSpareWheels(int spareWheels) {
		this.spareWheels = spareWheels;
	}

	public List<String> getCarEnhancements() {
		return carEnhancements;
	}

	public void setCarEnhancements(List<String> carEnhancements) {
		this.carEnhancements = carEnhancements;
	}

	public String getConsumeProvisionText() {
		return getResources().getString(R.string.useMedKit);
	}
	

	public String getProvisionsText() {
		return getResources().getString(R.string.medKits);
	}
	
	public String getCurrencyName() {
		return getResources().getString(R.string.credits);
	}

}

File: src/main/java/pt/joaomneto/titancompanion/adventure/impl/FFAdventure.kt
package pt.joaomneto.titancompanion.adventure.impl

import android.os.Bundle
import android.view.Menu
import pt.joaomneto.titancompanion.R
import pt.joaomneto.titancompanion.adventure.Adventure
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureEquipmentFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureNotesFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.AdventureVitalStatsFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.ff.FFAdventureCombatFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.ff.FFVehicleCombatFragment
import pt.joaomneto.titancompanion.adventure.impl.fragments.ff.FFVehicleStatsFragment
import pt.joaomneto.titancompanion.util.AdventureFragmentRunner
import java.io.BufferedWriter
import java.io.IOException
import java.util.ArrayList

class FFAdventure : Adventure(
    arrayOf(
        AdventureFragmentRunner(R.string.vitalStats, AdventureVitalStatsFragment::class),
        AdventureFragmentRunner(R.string.fights, FFAdventureCombatFragment::class),
        AdventureFragmentRunner(R.string.vehicleCombat, FFVehicleCombatFragment::class),
        AdventureFragmentRunner(R.string.vehicleSpecs, FFVehicleStatsFragment::class),
        AdventureFragmentRunner(R.string.goldEquipment, AdventureEquipmentFragment::class),
        AdventureFragmentRunner(R.string.notes, AdventureNotesFragment::class)
    )
) {

    var carEnhancements: List<String> = ArrayList()
    var currentFirepower = -1
    var currentArmour = -1
    var initialFirepower = -1
    var initialArmour = -1
    var rockets = -1
    var ironSpikes = -1
    var oilCannisters = -1
    var spareWheels = -1

    override val consumeProvisionText = R.string.useMedKit

    override val provisionsText = R.string.medKits

    override val currencyName = R.string.credits

    override fun onCreate(savedInstanceState: Bundle?) {
        try {
            super.onCreate(savedInstanceState)
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.adventure, menu)
        return true
    }

    @Throws(IOException::class)
    override fun storeAdventureSpecificValuesInFile(bw: BufferedWriter) {

        bw.write("currentFirepower=" + currentFirepower + "\n")
        bw.write("currentArmour=" + currentArmour + "\n")
        bw.write("initialFirepower=" + initialFirepower + "\n")
        bw.write("initialArmour=" + initialArmour + "\n")
        bw.write("rockets=" + rockets + "\n")
        bw.write("ironSpikes=" + ironSpikes + "\n")
        bw.write("oilCannisters=" + oilCannisters + "\n")
        bw.write("spareWheels=" + spareWheels + "\n")

        bw.write("gold=" + gold + "\n")
        bw.write("provisions=" + provisions + "\n")
        bw.write("provisionsValue=" + provisionsValue + "\n")
        bw.write("carEnhancements=" + Adventure.Companion.arrayToString(carEnhancements))
    }

    override fun loadAdventureSpecificValuesFromFile() {

        currentFirepower = Integer.valueOf(savedGame.getProperty("currentFirepower"))
        currentArmour = Integer.valueOf(savedGame.getProperty("currentArmour"))
        initialFirepower = Integer.valueOf(savedGame.getProperty("initialFirepower"))
        initialArmour = Integer.valueOf(savedGame.getProperty("initialArmour"))
        rockets = Integer.valueOf(savedGame.getProperty("rockets"))
        ironSpikes = Integer.valueOf(savedGame.getProperty("ironSpikes"))
        oilCannisters = Integer.valueOf(savedGame.getProperty("oilCannisters"))
        gold = Integer.valueOf(savedGame.getProperty("gold"))
        provisions = Integer.valueOf(savedGame.getProperty("provisions"))
        provisionsValue = Integer.valueOf(savedGame.getProperty("provisionsValue"))
        spareWheels = Integer.valueOf(savedGame.getProperty("spareWheels"))
        carEnhancements = stringToArray(savedGame.getProperty("carEnhancements"))
    }

    companion object {

        protected val FRAGMENT_VEHICLE_COMBAT = 2
        protected val FRAGMENT_VEHICLE_EQUIPMENT = 3
        protected val FRAGMENT_EQUIPMENT = 4
        protected val FRAGMENT_NOTES = 5
    }
}
